extensions/csrf-reactive/runtime/src/main/java/io/quarkus/csrf/reactive/runtime/CsrfRecorder.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.csrf.reactive.runtime;
+
+import org.jboss.resteasy.reactive.server.core.Deployment;
+import org.jboss.resteasy.reactive.server.spi.ServerRestHandler;
+
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class CsrfRecorder {
+
+    public void configure(RuntimeValue<Deployment> deployment, CsrfReactiveConfig csrfReactiveConfig) {
+        for (ServerRestHandler handler : deployment.getValue().getRuntimeConfigurableServerRestHandlers()) {
+            if (handler instanceof CsrfHandler) {
+                ((CsrfHandler) handler).configure(csrfReactiveConfig);
+                break;
```

Comment:
This `break` statement is causing most of the problems you are seeing.

You should **not**  break out of the loop because each JAX-RS method has its own set of handlers